enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -0,0 +1,192 @@
+        else:
+            raise ValueError(f"unknown access method {self.access_method}")
+
+
+class SubscriptionAccessPolicy(SubsidyAccessPolicy):
+    """
+    A subsidy access policy for subscriptions.
+    
+    .. no_pii:
+    """
+    class Meta:
+        """
+        Metaclass for SubscriptionAccessPolicy.
+        """
+        abstract = True
+    
+    def can_redeem(self, learner_id, content_key):
+        if subsidy_client.get_license_for_learner(self.subsidy_uuid, learner_id):
+            return super().can_redeem(learner_id, content_key)
+
+        group_uuids = {
+            group['uuid'] for group
+            in group_client.get_groups_for_learner(learner_id)
+        }
+        if self.group_uuid in group_uuids:
+            if subsidy_client.get_license_for_group(self.subsidy_uuid, self.group_uuid):
+                return super().can_redeem(learner_id, content_key)
+
+        return False
+
+
+class LearnerCreditAccessPolicy(SubsidyAccessPolicy):
+    """
+    A subsidy access policy for learner credit.
+    
+    .. no_pii:
+    """
+    class Meta:
+        """
+        Metaclass for LearnerCreditAccessPolicy.
+        """
+        abstract = True
+
+
+class LicenseRequestAccessPolicy(SubscriptionAccessPolicy):
+    """
+    A subsidy access policy for subscriptions with a request type access method.
+    """
+    access_method = AccessMethods.REQUEST
+
+
+class LicenseAccessPolicy(SubscriptionAccessPolicy):
+    """
+    A subsidy access policy for subscriptions with a direct type access method.
+    """
+    access_method = AccessMethods.DIRECT
+
+
+class PerLearnerEnrollmentCapLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Policy that limits the number of enrollments transactions for a learner in a subsidy.
+    """
+    per_learner_enrollment_limit = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+
+    def can_redeem(self, learner_id, content_key):
+        learner_transaction_count = subsidy_client.transactions_for_learner(
+            subsidy_uuid = self.subsidy_uuid,
+            learner_id = learner_id,
+            ).count()
+        if learner_transaction_count < self.per_learner_enrollment_limit:
+            return super().can_redeem(learner_id, content_key)
+        
+        return False
+
+class PerLearnerSpendCapLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Policy that limits the amount of learner spend for enrollment transactions.
+    """
+    per_learner_spend_limit = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+
+    def can_redeem(self, learner_id, content_key):
+        spent_amount = subsidy_client.amount_spent_for_learner(
+            subsidy_uuid = self.subsidy_uuid,
+            learner_uuid = learner_id,
+            )
+        #TODO the following should also incorporate the course price
+        if spent_amount < self.per_learner_spend_limit:
```

Comment:
@iloveagent57 This should account for the price of the course in the transaction ... right? Price to be fetched from which client?

Reply:
Yup, exactly.  This should be fetched from the Subsidy API/client.  I'll try to get you a real API specification soon so that this logic can be filled out a little more.